[libmad] Build error on x64-windows (wrong hash)

So... either the file on the server changed (supply-chain attack?) or the hash is simply wrong?

Probably related to e5e9f2f145e539a97b4a56c857892e89a74751e0 and issue #35898 

Here's the output of the `issue_body.md` file:

Package: libmad:x64-windows@0.16.4

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.38.33134.0
-    vcpkg-tool version: 2024-01-11-710a3116bbd615864eef5f9010af178034cb9b44
    vcpkg-scripts version: 0f9a4f65e 2024-01-25 (2 hours ago)

**To Reproduce**

`vcpkg install libmad`
**Failure logs**

```
-- Note: libmad only supports static library linkage. Building static library.
-- Downloading https://codeberg.org/tenacityteam/libmad/archive/0.16.4.tar.gz -> tenacityteam-libmad-0-6752c199.16.4.tar.gz...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Trying to load bundleconfig from d:\Devel\vcpkg\vcpkg-bundle.json
[DEBUG] Failed to open: d:\Devel\vcpkg\vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Metrics enabled.
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Feature flag 'dependencygraph' unset
Downloading https://codeberg.org/tenacityteam/libmad/archive/0.16.4.tar.gz
[DEBUG] Trying to hash D:\Devel\vcpkg\downloads\tenacityteam-libmad-0-6752c199.16.4.tar.gz.20768.part
[DEBUG] D:\Devel\vcpkg\downloads\tenacityteam-libmad-0-6752c199.16.4.tar.gz.20768.part has hash e56b84e112e3a95cd4fde7be1ce94c4a277275da3cc69ca051113162a84158ac9a1fe9c3f2de186202de5eca5c112aacb87533206288a084cf5ba048e6d95c22
error: Failed to download from mirror set
error: File does not have the expected hash:
url: https://codeberg.org/tenacityteam/libmad/archive/0.16.4.tar.gz
File: D:\Devel\vcpkg\downloads\tenacityteam-libmad-0-6752c199.16.4.tar.gz.20768.part
Expected hash: 6752c199096f999ed478dea712eb669913eec182eec8a56bf4871e77e6c38798d13146febf646007427f43f86aea8d40b9ec2b72928ac3132066ebddcaa0cfde
Actual hash: e56b84e112e3a95cd4fde7be1ce94c4a277275da3cc69ca051113162a84158ac9a1fe9c3f2de186202de5eca5c112aacb87533206288a084cf5ba048e6d95c22
[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(1029): 
[DEBUG] Time in subprocesses: 0us
[DEBUG] Time in parsing JSON: 3us
[DEBUG] Time in JSON reader: 0us
[DEBUG] Time in filesystem: 4720us
[DEBUG] Time in loading ports: 0us
[DEBUG] Exiting after 962 ms (860259us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
      
      Failed to download file with error: 1
      If you are using a proxy, please check your proxy setting. Possible causes are:
      
      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.
      
      2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
         set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
         The value set by your proxy might be wrong, or have same `https://` prefix issue.
      
      3. Your proxy's remote server is out of service.
      
      If you've tried directly download the link, and believe this is not a temporary
      download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
      to report this upstream download server failure.
      

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
  ports/libmad/portfile.cmake:5 (vcpkg_download_distfile)
  scripts/ports.cmake:170 (include)



```

--- a/ports/libmad/portfile.cmake
+++ b/ports/libmad/portfile.cmake
@@ -6,7 +6,7 @@ vcpkg_download_distfile(
     ARCHIVE
     URLS "https://codeberg.org/tenacityteam/libmad/archive/${VERSION}.tar.gz"
     FILENAME "tenacityteam-libmad-${VERSION}.tar.gz"
-    SHA512 6752c199096f999ed478dea712eb669913eec182eec8a56bf4871e77e6c38798d13146febf646007427f43f86aea8d40b9ec2b72928ac3132066ebddcaa0cfde
+    SHA512 e56b84e112e3a95cd4fde7be1ce94c4a277275da3cc69ca051113162a84158ac9a1fe9c3f2de186202de5eca5c112aacb87533206288a084cf5ba048e6d95c22
 )
 
 vcpkg_extract_source_archive(SOURCE_PATH ARCHIVE "${ARCHIVE}")

--- a/ports/libmad/portfile.cmake
+++ b/ports/libmad/portfile.cmake
@@ -2,11 +2,14 @@ if(VCPKG_TARGET_IS_WINDOWS)
     vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 endif()
 
-vcpkg_from_git(
-    OUT_SOURCE_PATH SOURCE_PATH
-    URL "https://codeberg.org/tenacityteam/libmad/"
-    REF ee7c3cdd4361b7f7c7da52fa3ba9f76cc35230a9
+vcpkg_download_distfile(
+    ARCHIVE
+    URLS "https://codeberg.org/tenacityteam/libmad/releases/download/${VERSION}/libmad-${VERSION}.tar.gz"
+    FILENAME "tenacityteam-libmad-${VERSION}.tar.gz"
+    SHA512 5b0a826408395e8b6b8a33953401355d6c2f1b33ec5085530b4ac8a538c39ffa903ce2e6845e9dcad73936933078959960b2f3fbba11ae091fda5bc5ee310df5
 )
+
+vcpkg_extract_source_archive(SOURCE_PATH ARCHIVE "${ARCHIVE}")
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"

--- a/ports/libmad/portfile.cmake
+++ b/ports/libmad/portfile.cmake
@@ -2,14 +2,11 @@ if(VCPKG_TARGET_IS_WINDOWS)
     vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 endif()
 
-vcpkg_download_distfile(
-    ARCHIVE
-    URLS "https://codeberg.org/tenacityteam/libmad/archive/${VERSION}.tar.gz"
-    FILENAME "tenacityteam-libmad-${VERSION}.tar.gz"
-    SHA512 e56b84e112e3a95cd4fde7be1ce94c4a277275da3cc69ca051113162a84158ac9a1fe9c3f2de186202de5eca5c112aacb87533206288a084cf5ba048e6d95c22
+vcpkg_from_git(
+    OUT_SOURCE_PATH SOURCE_PATH
+    URL "https://codeberg.org/tenacityteam/libmad/"
+    REF ee7c3cdd4361b7f7c7da52fa3ba9f76cc35230a9
 )
-
-vcpkg_extract_source_archive(SOURCE_PATH ARCHIVE "${ARCHIVE}")
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"

--- a/ports/libmad/vcpkg.json
+++ b/ports/libmad/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "libmad",
   "version": "0.16.4",
+  "port-version": 1,
   "description": "high-quality MPEG audio decoder",
   "homepage": "http://codeberg.org/tenacityteam/libmad",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4526,7 +4526,7 @@
     },
     "libmad": {
       "baseline": "0.16.4",
-      "port-version": 0
+      "port-version": 1
     },
     "libmagic": {
       "baseline": "5.45",
@@ -9016,13 +9016,13 @@
       "baseline": "1.0.9",
       "port-version": 0
     },
-    "vsgxchange": {
-      "baseline": "1.0.5",
-      "port-version": 1
-    },
     "vsgimgui": {
       "baseline": "0.1.0",
       "port-version": 0
+    },
+    "vsgxchange": {
+      "baseline": "1.0.5",
+      "port-version": 1
     },
     "vtk": {
       "baseline": "9.2.0-pv5.11.0",

--- a/versions/l-/libmad.json
+++ b/versions/l-/libmad.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "9fb81037e132bbcbb8dd5c239e347e9450ae888a",
+      "version": "0.16.4",
+      "port-version": 1
+    },
+    {
       "git-tree": "ca6bd50225347461a9d72269c1b7d082a691f31d",
       "version": "0.16.4",
       "port-version": 0

--- a/versions/l-/libmad.json
+++ b/versions/l-/libmad.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "d6f89685c79a9aae3feb24e19895cc855c06253f",
+      "git-tree": "7016c2c6c5e65a6420a176aa97696897bb134c40",
       "version": "0.16.4",
       "port-version": 1
     },

--- a/versions/l-/libmad.json
+++ b/versions/l-/libmad.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "9fb81037e132bbcbb8dd5c239e347e9450ae888a",
+      "git-tree": "d6f89685c79a9aae3feb24e19895cc855c06253f",
       "version": "0.16.4",
       "port-version": 1
     },

